### PR TITLE
front: Fix All signals button in Map Settings 

### LINKED ISSUE #2835: "All signals" display button isn't working
### What happened?

"All signals" display on/off button isn't taking with him the two other on/off button when ticked. It is actually not doing anything at the moment.
image

On the other hand, when all signals are displayed, the "All signals" button can be off.
image

### What did you expect to happen?

- the "all signals" button should tick or untick all the others with him (in the signals section only)
~- it should be placed in a way that make it different from the other buttons in the signals section of the settings~ *To be defined later on, we can keep it like now, as the first switch*
- if all the buttons except this one appear to be on, it should turn on automatically
- if this button is on and one of the other signals button is turned off, it should follow and be turned off automatically

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to the home page of the time table module
2. Open the map settings
3. Try to tick or untick the "All signals" on/off button.

### What operating system and browser are you using? (if applicable)

- MSDOS 5
- Arachne 1.99

### OSRD version

1.0

### PR DESCRIPTION
**Pull request by Uriel-Sautron**
close #2835

### CODE REVIEW

File: front/src/common/Map/Settings/MapSettingsSignals.js [hunk 1]
```diff
@@ -16,13 +16,48 @@ export default function MapSettingsSignals() {
...
+      );
+    } else {
+      dispatch(
+        updateSignalsSettings({
+          ...signalsSettings,
+          [setting]: !signalsSettings[setting],
+        })
+      );
+    }
   };
```
Comment (anisometropie) [inline front/src/common/Map/Settings/MapSettingsSignals.js:49]: As we saw together :
```
  const setSignalsList = (setting) => {
    let newSetting 
    if (setting === 'all') {
      newSetting = switchAllSettings(setting, signalsSettings);
    } else if (checkedSettings(setting, signalsSettings)) {
      newSetting = switchAllSettings(setting, signalsSettings);
    } else {
      newSetting = {
        ...signalsSettings,
        [setting]: !signalsSettings[setting],
      }
    }
    dispatch(
      updateSignalsSettings(newSetting)
    );
  };
```

Comment (Uriel-Sautron) [reply front/src/common/Map/Settings/MapSettingsSignals.js:49]: Done

File: front/src/common/Map/Settings/MapSettingsSignals.js [hunk 2]
```diff
@@ -16,13 +16,38 @@ export default function MapSettingsSignals() {
     lights: lightsIcon,
   };
 
+  const switchAllSettings = (settingSwitch, stateOfSettings) => {
+    const allSettings = {};
+
+    CONSTS_SETTINGS.forEach(
+      (settingKey) => (allSettings[settingKey] = !stateOfSettings[settingSwitch])
```
Comment (anisometropie) [inline front/src/common/Map/Settings/MapSettingsSignals.js]: written like this, it seems like you want to use the value returned.

Here we just want to use the side-effect (adding a key to the object). 

you can simply do this :

```
    CONSTS_SETTINGS.forEach(
      (settingKey) => {
         allSettings[settingKey] = !stateOfSettings[settingSwitch]
      }
   );
```

File: front/src/common/Map/Settings/MapSettingsSignals.js [hunk 3]
```diff
@@ -16,13 +16,38 @@ export default function MapSettingsSignals() {
...
+    return settingValues.every((settingValue) => settingValue === !stateOfSettings[settingSwitch]);
+  };
+
   const setSignalsList = (setting) => {
-    dispatch(
-      updateSignalsSettings({
+    let newSettings;
+
+    if (setting === 'all') {
+      newSettings = switchAllSettings(setting, signalsSettings);
```
Comment (anisometropie) [inline front/src/common/Map/Settings/MapSettingsSignals.js]: To go one step furthere in the refactoring, we can see that the first two blocks are doing the same thing :

`newSettings = switchAllSettings(setting, signalsSettings);`

you can regroup in one single block :

```
if (setting === 'all' || checkedSettings(setting, signalsSettings)) {
  newSettings = switchAllSettings(setting, signalsSettings);
} else { 
…
}
```

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/src/common/Map/Settings/MapSettingsSignals.js (hunk 3 above)
Comment (Uriel-Sautron) [reply front/src/common/Map/Settings/MapSettingsSignals.js]: Thanks, done

**Review by anisometropie — APPROVED** [review]

**Pull request merged**